app/controllers/integrations_controller.rb

```diff
@@ -0,0 +1,18 @@
+class IntegrationsController < ApplicationController
+  load_and_authorize_resource
+
+  def update
+    if params[:application]
+      @integration.configuration["selected_application"] = @integration.configuration["selected_application"] = @integration.get_applications.find{|app| app[:title] == params[:application]}[:title]
```

Comment:
I don't like this. We need to create a method to select the application we want in the integrations code. Don't modify the configuration hash in the controllers.